### PR TITLE
feat: verify that the Project was not deleted before creating new env…

### PR DESCRIPTION
**Pull request by otlci**
Issue #, if available: GALI-2283

Description of changes:

* Verify that the Project was not deleted before creating new Environment 
* Verify that the Project was not deleted before associating ETC with project

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts [hunk 1]
```diff
@@ -53,4 +60,69 @@ describe('environment start negative tests', () => {
...
+      storageName: settings.get('DataSetsBucketName'),
+      awsAccountId: settings.get('mainAccountId'),
+      path: dataSetName, // using same name to help potential troubleshooting
+      name: dataSetName,
+      region: settings.get('awsRegion'),
+      owner: getProjectAdminRole(createdProject.id),
+      ownerType: 'GROUP',
+      type: 'internal',
+      permissions: [
+        {
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts:91]: do you need this permission for this test?

File: solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts [hunk 2]
```diff
@@ -53,4 +60,69 @@ describe('environment start negative tests', () => {
...
+      description: 'test description',
+      costCenterId: costCenter.id
+    });
+
+    const projectId = createdProject.id;
+
+    const dataSetBody = CreateDataSetRequestParser.parse({
+      storageName: settings.get('DataSetsBucketName'),
+      awsAccountId: settings.get('mainAccountId'),
+      path: dataSetName, // using same name to help potential troubleshooting
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts:84]: still need this comment?

File: solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts [hunk 3]
```diff
@@ -53,4 +60,69 @@ describe('environment start negative tests', () => {
...
+        {
+          identity: getResearcherRole(createdProject.id),
+          identityType: 'GROUP',
+          accessLevel: 'read-only'
+        }
+      ]
+    });
+
+    const { data: dataSet } = await adminSession.resources.datasets.create(dataSetBody, false);
+    expect(dataSet).toMatchObject({
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts:100]: Is this expectation related to the thing you're testing?

File: solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/create.test.ts [hunk 4]
```diff
@@ -68,6 +75,67 @@ describe('Associate Project with EnvTypeConfig', () => {
...
+
+    const { data: createdProject } = await adminSession.resources.projects.create({
+      name: `${dataSetName} project`,
+      description: 'test description',
+      costCenterId: costCenter.id
+    });
+
+    const projectId = createdProject.id;
+
+    const dataSetBody = CreateDataSetRequestParser.parse({
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/create.test.ts:95]: do you need a dataset for this test?

**Review by aws-tyler — APPROVED** [review]

File: solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts (hunk 1 above)
Comment (otlci) [reply solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts:91]: TBH I'm not sure, I've copied this test from other location and kept it intact to make sure it's working properly. Since the comments are addressing tests Is it ok to resolve the comments after code freeze as low risk changes?

File: solutions/swb-app/src/projectEnvTypeConfigRoutes.ts [hunk 5]
```diff
@@ -47,7 +48,14 @@ export function setUpProjectEnvTypeConfigRoutes(
...
         }
       );
-      await projectEnvTypeConfigService.associateProjectWithEnvTypeConfig(request);
+      try {
+        await projectEnvTypeConfigService.associateProjectWithEnvTypeConfig(request);
+      } catch (e) {
+        if (isProjectDeletedError(e)) {
+          throw Boom.badRequest(e.message);
+        }
+        throw e;
```
Comment (maghirardelli) [inline solutions/swb-app/src/projectEnvTypeConfigRoutes.ts:57]: We still want the route to throw a `Boom` error and just throwing `e` does not ensure that. Do this instead:
```
if (Boom.isBoom(e)) {
  throw e;
}
throw Boom.badImplementation(<helpfulMessage>);

```

File: solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts (hunk 3 above)
Comment (otlci) [reply solutions/swb-reference/integration-tests/tests/isolated/environments/start.test.ts:100]: No, it's from other test. I can remove that.

File: solutions/swb-app/src/projectEnvTypeConfigRoutes.ts [hunk 6]
```diff
@@ -47,7 +48,14 @@ export function setUpProjectEnvTypeConfigRoutes(
...
+      }
       res.status(201).send();
```
Comment (maghirardelli) [inline solutions/swb-app/src/projectEnvTypeConfigRoutes.ts:59]: This send should be in the `try` block

File: solutions/swb-app/src/projectEnvironmentRoutes.ts [hunk 7]
```diff
@@ -30,7 +31,15 @@ export function setUpProjectEnvRoutes(
         if (req.body.id) {
           throw badRequest('id cannot be passed in the request body when trying to launch a new environment');
         }
-        const env: Environment = await projectEnvironmentService.createEnvironment(req.body, res.locals.user);
+        let env: Environment;
+        try {
+          env = await projectEnvironmentService.createEnvironment(req.body, res.locals.user);
+        } catch (e) {
+          if (isProjectDeletedError(e)) {
```
Comment (maghirardelli) [inline solutions/swb-app/src/projectEnvironmentRoutes.ts:38]: same comment as above for catching and throwing errors

File: solutions/swb-app/src/projectEnvTypeConfigRoutes.ts (hunk 5 above)
Comment (aws-tyler) [reply solutions/swb-app/src/projectEnvTypeConfigRoutes.ts:57]: We should try to make sure any code changes use Boom :-D

**Review by maghirardelli — APPROVED** [review]
